### DISCUSSION TITLE
Is it possible to run tests?

### DISCUSSION THREAD

**Discussion by plondon** [codesandbox/sandpack#51] [Q&A] [answered]
Codesandbox has a wonderful integrated test runner that seems to automagically detect spec files. Is something like this possible with sandpack?

**Comment by plondon** [comment, marked as answer]
I'll just try to follow this thread and close this for now! https://github.com/codesandbox/codesandbox-client/issues/364

**Comment by CompuIves** [comment]
Hey! That's right! We have the same logic in Sandpack, but we're not using it right now. However, you should be able to trigger it by sending a message to sandpack. The message should be something like this: `{ "type": "run-all-tests" }`. You will get a plethora of messages back, mostly being status updates on the tests. At the end you should have a result.

Right now it's a bit wacky because we don't have official support within Sandpack for tests, but the logic is already there 😄 .

**Comment by gdimitropoulos** [comment]
@CompuIves  Is there any change on  sandpack so test would be triggered with dispatch({ type: "run-tests"}) ?
